plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlJdbcConnectionCreation.java

```diff
@@ -0,0 +1,153 @@
+        return createPostgreSqlQueryRunner(postgreSqlServer, ImmutableList.of(NATION, REGION), connectionFactory);
+    }
+
+    @Test(dataProvider = "testCases")
+    public void testJdbcConnectionCreations(@Language("SQL") String query, int expectedJdbcConnectionsCount, Optional<String> errorMessage)
+    {
+        assertJdbcConnections(query, expectedJdbcConnectionsCount, errorMessage);
+    }
+
+    @DataProvider
+    public Object[][] testCases()
+    {
+        return new Object[][] {
+                {"SELECT * FROM nation LIMIT 1", 5, Optional.empty()},
+                {"SELECT * FROM nation ORDER BY nationkey LIMIT 1", 5, Optional.empty()},
+                {"SELECT * FROM nation WHERE nationkey = 1", 5, Optional.empty()},
+                {"SELECT avg(nationkey) FROM nation", 4, Optional.empty()},
+                {"SELECT * FROM nation, region", 6, Optional.empty()},
+                {"SELECT * FROM nation n, region r WHERE n.regionkey = r.regionkey", 8, Optional.empty()},
+                {"SELECT * FROM nation JOIN region USING(regionkey)", 10, Optional.empty()},
+                {"SELECT * FROM information_schema.schemata", 1, Optional.empty()},
+                {"SELECT * FROM information_schema.tables", 1, Optional.empty()},
+                {"SELECT * FROM information_schema.columns", 5, Optional.empty()},
+                {"SELECT * FROM nation", 3, Optional.empty()},
+                {"SELECT * FROM TABLE (system.query(query => 'SELECT * FROM tpch.nation'))", 2, Optional.empty()},
+                {"CREATE TABLE copy_of_nation AS SELECT * FROM nation", 15, Optional.empty()},
+                {"INSERT INTO copy_of_nation SELECT * FROM nation", 13, Optional.empty()},
+                {"DELETE FROM copy_of_nation WHERE nationkey = 3", 4, Optional.empty()},
+                {"UPDATE copy_of_nation SET name = 'POLAND' WHERE nationkey = 1", 2, Optional.of("This connector does not support updates")},
+                {"MERGE INTO copy_of_nation n USING region r ON r.regionkey= n.regionkey WHEN MATCHED THEN DELETE", 2, Optional.of("This connector does not support merges")},
+                {"DROP TABLE copy_of_nation", 3, Optional.empty()},
+                {"SHOW SCHEMAS", 1, Optional.empty()},
+                {"SHOW TABLES", 2, Optional.empty()},
+                {"SHOW STATS FOR nation", 4, Optional.empty()},
+        };
+    }
+
+    private static DistributedQueryRunner createPostgreSqlQueryRunner(
+            TestingPostgreSqlServer server,
+            Iterable<TpchTable<?>> tables,
+            ConnectionCountingConnectionFactory connectionCountingConnectionFactory)
+            throws Exception
+    {
+        DistributedQueryRunner queryRunner = null;
+        try {
+            queryRunner = DistributedQueryRunner.builder(createSession())
+                    .setCoordinatorProperties(ImmutableMap.of("node-scheduler.include-coordinator", "false"))
+                    .build();
```

Comment:
is this significant for the test? Deserves some comment IMO.

Reply:
I wanted to make sure that we always open connections in a deterministic manner. Like we read data on worker. I don't think it is needed and possibly it lowers the test coverage. So I will remove it.